src/ui/src/components/core/content/CoreChatbot.vue

```diff
@@ -384,17 +467,139 @@ function handleAttachFiles() {
 }
 
 function handleRemoveFile(index: number) {
-	const newList = files.value.toSpliced(index, 1);
+	const newList = [...files.value];
+	newList.splice(index, 1);
 	files.value = newList;
 }
 
+async function handlePaste(event: ClipboardEvent) {
+	// Check if image pasting is enabled
+	if (fields.enableImagePaste.value !== true) return;
+
+	const items = event.clipboardData?.items;
+	if (!items) return;
+
+	const imageItems = [];
+	for (let i = 0; i < items.length; i++) {
+		const item = items[i];
+		if (item.type.startsWith("image/")) {
+			const file = item.getAsFile();
+			if (file) {
+				imageItems.push(file);
+			}
+		}
+	}
+
+	if (imageItems.length === 0) return;
+
+	// Prevent default paste for images
+	event.preventDefault();
+
+	// Show immediate visual feedback with placeholder URLs
+	const placeholderImages = imageItems.map(
+		() =>
+			"data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTIwIiBoZWlnaHQ9IjgwIiB2aWV3Qm94PSIwIDAgMTIwIDgwIiBmaWxsPSJub25lIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPjxyZWN0IHdpZHRoPSIxMjAiIGhlaWdodD0iODAiIGZpbGw9IiNmM2Y0ZjYiLz48Y2lyY2xlIGN4PSI2MCIgY3k9IjQwIiByPSIxNiIgZmlsbD0iIzlca2E0YWYiPjxhbmltYXRlIGF0dHJpYnV0ZU5hbWU9Im9wYWNpdHkiIHZhbHVlcz0iMC4yOzE7MC4yIiBkdXI9IjEuNXMiIHJlcGVhdENvdW50PSJpbmRlZmluaXRlIi8+PC9jaXJjbGU+PC9zdmc+",
+	);
+	pastedImages.value = [...pastedImages.value, ...placeholderImages];
+	processingImages.value = true;
+
+	// Process images in the background
+	try {
+		await Promise.all(
+			imageItems.map(async (file, index) => {
+				try {
+					// Optimize image if it's too large
+					const optimizedFile = await optimizeImage(file);
+					const dataUrl = await encodeFile(optimizedFile);
+
+					// Replace placeholder with actual image
+					const currentImages = [...pastedImages.value];
+					const placeholderIndex =
+						currentImages.length - imageItems.length + index;
+					currentImages[placeholderIndex] = dataUrl as string;
+					pastedImages.value = currentImages;
+
+					return dataUrl as string;
+				} catch (error) {
+					logger.error("Failed to process pasted image:", error);
+					// Remove the placeholder if processing fails
+					const currentImages = [...pastedImages.value];
+					const placeholderIndex =
+						currentImages.length - imageItems.length + index;
+					currentImages.splice(placeholderIndex, 1);
+					pastedImages.value = currentImages;
+					return null;
+				}
+			}),
+		);
+	} catch (error) {
+		logger.error("Error processing pasted images:", error);
+	} finally {
+		processingImages.value = false;
+	}
+}
+
+function handleRemovePastedImage(index: number) {
+	const newList = [...pastedImages.value];
+	newList.splice(index, 1);
+	pastedImages.value = newList;
+}
+
 function scrollToBottom() {
 	messageAreaEl.value.scrollTo({
 		top: messageAreaEl.value.scrollHeight,
 		left: 0,
 	});
 }
 
+const optimizeImage = async (file: File): Promise<File> => {
```

Comment:
Maybe this is worth moving into a new utils `src/ui/src/utils/img.ts`. It helps to reduce this file size and make it clearer that it's an independent function.